tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -136,14 +135,17 @@ public void lineageCompleteAndAsyncPersistTest() throws Exception {
       HeartbeatScheduler.schedule(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING);
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING, 5,
           TimeUnit.SECONDS));
+      status = getFileSystemMasterClient().getStatus(uri);
       HeartbeatScheduler.schedule(HeartbeatContext.WORKER_FILESYSTEM_MASTER_SYNC);
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_FILESYSTEM_MASTER_SYNC, 5,
           TimeUnit.SECONDS));
-
-      status = getFileSystemMasterClient().getStatus(uri);
       Assert.assertEquals(PersistenceState.IN_PROGRESS.toString(), status.getPersistenceState());
+      CommonUtils.sleepMs(500);
```

Comment:
So you are saying you have to sleep enough time so the worker finishes something before the heartbeat is scheduled below? 500ms is no guarantee that it finishes so this could be a source of flaky tests, right?